File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/MarvelCharacters.java

package com.noblemajesty.marvel.models.getCharacters;

import android.os.Parcel;
import android.os.Parcelable;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class MarvelCharacters implements Parcelable
{

  @SerializedName("code")
  @Expose
  private Integer code;
  @SerializedName("data")
  @Expose
  private Data data;
  public final static Parcelable.Creator<MarvelCharacters> CREATOR = new Creator<MarvelCharacters>() {


    @SuppressWarnings({
        "unchecked"
    })
    public MarvelCharacters createFromParcel(Parcel in) {
      return new MarvelCharacters(in);
    }

    public MarvelCharacters[] newArray(int size) {
      return (new MarvelCharacters[size]);
    }

  };

  protected MarvelCharacters(Parcel in) {
    this.code = ((Integer) in.readValue((Integer.class.getClassLoader())));
    this.data = ((Data) in.readValue((Data.class.getClassLoader())));
  }

  public MarvelCharacters() {
  }

  public Integer getCode() {
    return code;
  }

  public void setCode(Integer code) {
    this.code = code;
  }

  public Data getData() {
    return data;
  }

  public void setData(Data data) {
    this.data = data;
  }

  public void writeToParcel(Parcel dest, int flags) {
    dest.writeValue(data);
  }

  public int describeContents() {
    return 0;
  }

}

File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/MarvelCharacters.kt
package com.noblemajesty.marvel.models.getCharacters

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class MarvelCharacters : Parcelable {

    @SerializedName("code")
    @Expose
    var code: Int? = null
    @SerializedName("data")
    @Expose
    var data: Data? = null

    protected constructor(`in`: Parcel) {
        this.code = `in`.readValue(Int::class.java.classLoader) as Int
        this.data = `in`.readValue(Data::class.java.classLoader) as Data
    }

    constructor() {}

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(data)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<MarvelCharacters> = object : Parcelable.Creator<MarvelCharacters> {


            override fun createFromParcel(`in`: Parcel): MarvelCharacters {
                return MarvelCharacters(`in`)
            }

            override fun newArray(size: Int): Array<MarvelCharacters?> {
                return arrayOfNulls(size)
            }

        }
    }

}
